src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -282,10 +293,32 @@ private Voucher withRequiredFields(Voucher voucher, Invoice invoice) {
     Voucher voucher = new Voucher();
     voucher.setInvoiceId(invoice.getId());
 
-    return voucherHelper.generateVoucherNumber()
+    return getVoucherNumberWithPrefix()
       .thenApply(voucher::withVoucherNumber);
   }
 
+  private CompletableFuture<String> getVoucherNumberWithPrefix() {
+    return voucherHelper.generateVoucherNumber()
```

Comment:
According to jira description voucher number prefix validation should be done prior to sequence generation

Reply:
Done